sdk/servicebus/service-bus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,248 @@
+
+#### "TransportType=AmqpWebSockets" Alternative
+
+To configure web socket use, see: [Use Proxy sample][ProxySample].
+
+#### "Authentication=Managed Identity" Alternative
+
+To authenticate with Managed Identity, see: [Use AAD Auth sample][IdentitySample]
+
+For more information about the `Azure.Identity` library, see: [Authentication and the Azure SDK][AuthenticationAndTheAzureSDK].
+
+## Logging and diagnostics
+
+The Service Bus client library is fully instrumented for logging information at various levels of detail.  Logging is performed for each operation and follows the pattern of marking the starting point of the operation, it's completion, and any errors encountered. Additional information that may offer insight is also logged in the context of the associated operation.
+
+### Enable logging
+
+You can set the following environment variable to get the debug logs when using this library.
+
+- Getting debug logs from the Service Bus SDK
+
+```bash
+export DEBUG=azure*
+```
+
+- Getting debug logs from the Service Bus SDK and the protocol level library.
+
+```bash
+export DEBUG=azure*,rhea*
+```
+
+- If you are **not interested in viewing the message transformation** (which consumes lot of console/disk space) then you can set the `DEBUG` environment variable as follows:
+
+```bash
+export DEBUG=azure*,rhea*,-rhea:raw,-rhea:message,-azure:core-amqp:datatransformer
+```
+
+- If you are interested only in **errors**, then you can set the `DEBUG` environment variable as follows:
+
+```bash
+export DEBUG=azure:service-bus:error,azure:core-amqp:error,rhea-promise:error,rhea:events,rhea:frames,rhea:io,rhea:flow
+```
+
+### Logging to a file
+
+1. Set the `DEBUG` environment variable as shown above
+2. Run your test script as follows:
+
+- Logging statements from your test script go to `out.log` and logging statements from the sdk go to `debug.log`.
+  ```bash
+  node your-test-script.js > out.log 2>debug.log
+  ```
+- Logging statements from your test script and the sdk go to the same file `out.log` by redirecting stderr to stdout (&1), and then redirect stdout to a file:
+  ```bash
+  node your-test-script.js >out.log 2>&1
+  ```
+- Logging statements from your test script and the sdk go to the same file `out.log`.
+  ```bash
+    node your-test-script.js &> out.log
+  ```
+### Distributed tracing
+
+The Service Bus client library has **experimental** support for the the OpenTelemetry specification via the [`@azure/opentelemetry-instrumentation-azure-sdk` package](https://www.npmjs.com/package/@azure/opentelemetry-instrumentation-azure-sdk).
+
+The library creates the following spans:
+
+`message`
+`ServiceBusSender.send`
+`StreamReceiver.process`
+`BatchingReceiverLite.process`
+`SessionReceiver.process`
+`ServiceBusRuleManager.createRule`
+`ServiceBusRuleManager.deleteRule`
+`ServiceBusRuleManager.getRules`
```

Comment:
we have a work item tracking support for other operations like defer, cancel, etc.